float32 left
bool left_change
float32 center
bool center_change
float32 right
bool right_change